exonum-java-binding/core/src/main/java/com/exonum/binding/core/runtime/ServiceArtifactId.java

```diff
@@ -58,16 +63,20 @@ public static ServiceArtifactId parseFrom(String serviceArtifactId) {
     String[] coordinates = serviceArtifactId.split(DELIMITER, NUM_FIELDS);
     int runtimeId = parseInt(coordinates[0]);
     String name = coordinates[1];
-    return valueOf(runtimeId, name);
+    String version = coordinates[2];
+    return valueOf(runtimeId, name, version);
   }
 
   /**
    * Creates a new service artifact id of a Java artifact.
    *
-   * @param name the name of the service; must not be blank
+   * @param artifactName the name of the artifact; must not be blank
    */
-  public static ServiceArtifactId newJavaId(String name) {
-    return valueOf(JAVA.getId(), name);
+  public static ServiceArtifactId newJavaId(String artifactName) {
```

Comment:
It makes no sense: it accepts the _name_, which is `'groupId<delimiter>artifactId'`
  — where from do you pull the version? There must not be 'artifactName' that is not the same as ArtifactId#getName.

Second, the delimiter.